normandy/recipes/tests/test_models.py

```diff
@@ -420,6 +410,36 @@ def test_signatures_update_correctly_on_enable(self, mocked_autograph):
         assert recipe.signature is not None
         assert recipe.signature.signature == fake_sign([recipe.canonical_json()])[0]['signature']
 
+    def test_only_signed_when_approved_and_enabled(self, mocked_autograph):
+        sign_data_mock = mocked_autograph.return_value.sign_data
+        # This uses the signer, so do it first
+        action = ActionFactory()
+        sign_data_mock.reset_mock()
+
+        sign_data_mock.side_effect = Exception("Can't sign yet")
+        recipe = RecipeFactory(name='unchanged', action=action)
+        assert not recipe.enabled
+        assert not recipe.is_approved
+        assert recipe.signature is None
+
+        # Updating does not generate a signature
+        recipe.revise(name='changed')
+        assert recipe.signature is None
```

Comment:
The above comment about the lack of `.save()` should get caught here if you do `recipe.refresh_from_db()`, no?

Reply:
This is a good point. I'll add some extra checks to these tests to make sure that changes are being persisted correctly.